packages/cst-dts-gen/src/api.ts

```diff
@@ -0,0 +1,25 @@
+import { BaseParser } from "chevrotain"
+import { GenerateDtsOptions } from "../api"
+import { buildModel } from "./model"
+import { genDts, GenDtsOptions } from "./generate"
+
+const defaultOptions: GenDtsOptions = {
+  includeTypes: true,
+  includeVisitorInterface: true,
+  visitorInterfaceName: "ICstNodeVisitor"
+}
+
+export function generateCstDts(
+  parser: BaseParser,
+  options?: GenerateDtsOptions
+): string {
+  const effectiveOptions: GenDtsOptions = {
+    ...defaultOptions,
+    ...options
+  }
+
+  const productions = parser.getGAstProductions()
```

Comment:
we may want to only pass the `getGAstProduction[]` here instead of the whole parser instance

Reply:
Yeah, I considered that at first.  But thought that having the api accept a parser would be easier on the user to understand.  And possibly easier to extend if we needed more off the parser in the future.

But just passing the the GAST productions is fine to.